auslib/db.py

```diff
@@ -526,9 +526,26 @@ def forUpdate(self, rowData, changed_by):
         row['timestamp'] = self.getTimestamp()
         return self._insertStatement(**row)
 
-    def getChange(self, change_id, transaction=None):
-        """ Returns the unique change that matches the give change_id """
-        changes = self.select(where=[self.change_id == change_id], transaction=transaction)
+    def getChange(self, change_id=None, column_values=None, data_version=None, transaction=None):
+        """ Returns the unique change that matches the give change_id or
+            combination of data_version and values for the specified columns.
+            column_values is a dict that contains the column names that are
+            versioned and their values."""
+        if change_id is None:
+            where = [self.data_version == data_version]
+            column_names = {col.name: col for col in self.table.columns if col.name in self.base_primary_key}
+            for col in column_names.keys():
+                if col in column_values.keys():
+                    where.append(column_names[col] == column_values[col])
+                else:
+                    raise ValueError("Entire primary key not present")
+            if self.baseTable.versioned:
+                changes = self.select(where=where,
+                                      transaction=transaction)
+            else:
+                raise ValueError("data_version queried for non-versioned table")
+        else:
+            changes = self.select(where=[self.change_id == change_id], transaction=transaction)
```

Comment:
Does this mean that you can only retrieve history by change_id for non-versioned tables? That's probably OK for now, but please add a comment to make note of it.

I would like to see the arg checking happen earlier though. Eg: before the "if change_id" - it's a bit tidier this way. Should be ready to merge with that fixed.